README.md
<!-- README.md is generated from README.Rmd. Please edit that file -->
This is a package for making easy maps of Denmark.

Currently, the package allows you to do three things:

1.  make basic maps of Denmark
2.  turn these maps into (static) choropleth maps
3.  plot longitude-latitude points on a map of Denmark

I recommend you install the development version from github, using Hadley Wickham's [devtools](http://cran.r-project.org/web/packages/devtools/index.html) package:

    if(!require("devtools")) install.packages("devtools")
    devtools::install_github("sebastianbarfort/mapDK")

To create a basic map of Denmark at the municipality level simply run

``` r
library(mapDK)
mapDK()
```

![](README-unnamed-chunk-2-1.png) 

You can control the level of the map by specifying the `detail` argument. Currenty, `mapDK` accepts the following arguments

-   `municipality` - plots Denmark's 98 municipalities
-   `region` - plots Denmark's 5 regions
-   `rural` - plots Denmark's 11 rural areas
-   `zip` - plots Denmark's 598 zip code areas
-   `polling` - plots Denmark's 1385 polling places (as of 2015)
-   `parish` - plots Denmark's 1931 parishes

Choropleth Maps
---------------

`mapDK` also allows you to create static choropleth maps using either the package's build in datasets, or new datasets provided by the user.

### Plot Municipality Level Data

To create a static choropleth map at the municipality level we can use some data about burglaries in Denmark (included in the package).

One creates a choropleth map simply by specifying the values and id's (as strings) and the dataset in the call to `mapDK`

``` r
mapDK(values = "indbrud", id = "kommune", data = crime)
```

![](README-unnamed-chunk-3-1.png) 

If you don't provide names for all municipalities, the function will throw a warning.

Plot Polling Place Data
-----------------------

You can create cool maps of election results at the polling place level by specifying `detail = "polling"`. The package includes a dataset of Danish general election 2011 results at the polling level. Below, we plot Socialdemokratiets election results in percent at each available polling place in the map dataset

``` r
mapDK(values = "stemmer", id = "id", 
  data = subset(votes, navn == "socialdemokratiet"),
  detail = "polling", show_missing = FALSE,
  guide.label = "Stemmer \nSocialdemokratiet (pct)")
```

![](README-unnamed-chunk-4-1.png) 

Say you only want to plot Socialdemokratiets votes in Copenhagen. That's easily done using the `sub.plot` option

``` r
mapDK(values = "stemmer", id = "id", 
  data = subset(votes, navn == "socialdemokratiet"),
  detail = "polling", show_missing = FALSE,
  guide.label = "Stemmer \nSocialdemokratiet (pct)",
  sub.plot = "koebenhavn")
```

![](README-unnamed-chunk-5-1.png) 

Plot points on a map of Denmark
-------------------------------

You can provide your own data set of longitude-latitude points (in WGS84) and plot them on a map of Denmark using the `pointDK` function. Below, I plot a data set of benches in Copenhagen available in the package

``` r
pointDK(benches, sub = "koebenhavn", point.colour = "red")
```

![](README-unnamed-chunk-6-1.png) 

You can also plot values in your data by specifying a `values` column

``` r
benches$mydata = 1:nrow(benches) # create values
pointDK(benches, values = "mydata", detail = "polling", sub.plot = "koebenhavn", point.colour = "red",
       aesthetic = "colour")
```

![](README-unnamed-chunk-7-1.png) 

The getID function
------------------

The `getID` function allows you to print the map key. Running this before plotting your dataset with `mapDK` is probably a good idea.

`getID` accepts only one argument, `detail`, and using it is as easy as (it returns keys for municipalities if nothing else is specified)

``` r
getID()[1:10]
#>  [1] "aabenraa"    "aalborg"     "aeroe"       "albertslund" "alleroed"   
#>  [6] "aarhus"      "assens"      "ballerup"    "billund"     "bornholm"
```

Say you want the names of the parishes instead, just run `mapDK(detail = "parish")`.

`mapDK` and `ggmap`
-------------------

The functions in `mapDK` work nicely with `ggmap`. Below is an example

``` r
library("ggmap")
library("dplyr")
votes.cph.shape = mapDK::polling %>% filter(KommuneNav == "koebenhavn") %>% left_join(mapDK::votes)
cph.map = ggmap(get_map(location = c(12.57, 55.68), 
                       source = "stamen", 
                       maptype = "toner", crop = TRUE,
                       zoom = 13))
cph.map + 
  geom_polygon(data = subset(votes.cph.shape, navn == "socialdemokratiet"), 
                       aes(x = long, y = lat,
                           group = group, fill = stemmer),
                       alpha = .75) 
```

![](README-unnamed-chunk-9-1.png)


NEWS.md

# mapdk 0.4.0

## Plot points using `pointDK`

- New `pointDK` function allows you to plot longitude-latitude points on a map of Denmark

## Bug fixes and improvements

- all data sets in the package now use coordinates in WGS84
- better defaults (background now gray)


R/mapDK-package.R
#' Data frame of Denmark's 98 municipalities
#'
#' A dataset containing the longitude and latitude of all of Denmark's 98 municipalities
#'
#' @format A data frame with 39184 rows and 4 variables:
#' \describe{
#'   \item{long}{Longitude of municipality}
#'   \item{lat}{Latitude of municipality}
#'   \item{group}{Groups, generated by \code{ggmap:fortify}}
#'   \item{id}{Name of municipality}
#' }
#' @source \url{Statistics Denmark}
"municipality"
#' Data frame of Denmark's 5 regions
#'
#' A dataset containing the longitude and latitude of all of Denmark's 5 regions
#'
#' @format A data frame with 32522 rows and 4 variables:
#' \describe{
#'   \item{long}{Longitude of region}
#'   \item{lat}{Latitude of region}
#'   \item{group}{Groups, generated by \code{ggmap:fortify}}
#'   \item{id}{Name of region}
#' }
#' @source \url{Statistics Denmark}
"region"
#' Data frame of Denmark's 11 rural areas
#'
#' A dataset containing the longitude and latitude of all of Denmark's 11 rural areas
#'
#' @format A data frame with 4083 rows and 4 variables:
#' \describe{
#'   \item{long}{Longitude of rural area}
#'   \item{lat}{Latitude of rural area}
#'   \item{group}{Groups, generated by \code{ggmap:fortify}}
#'   \item{id}{Name of rural area}
#' }
#' @source \url{Statistics Denmark}
"rural"
#' Data frame of Denmark's 598 zip code areas
#'
#' A dataset containing the longitude and latitude of all of Denmark's 598 zip oode areas
#'
#' @format A data frame with 49322 rows and 4 variables:
#' \describe{
#'   \item{long}{Longitude of zip code area}
#'   \item{lat}{Latitude of zip code area}
#'   \item{group}{Groups, generated by \code{ggmap:fortify}}
#'   \item{id}{Identification number of zip code area}
#' }
#' @source \url{Statistics Denmark}
"zip"
#' Data frame of Denmark's 1931 parishes
#'
#' A dataset containing the longitude and latitude of all of Denmark's 1931 parishes
#'
#' @format A data frame with 76156 rows and 4 variables:
#' \describe{
#'   \item{long}{Longitude of parish}
#'   \item{lat}{Latitude of parish}
#'   \item{group}{Groups, generated by \code{ggmap:fortify}}
#'   \item{id}{Identification number of parish}
#' }
#' @source \url{Statistics Denmark}
"parish"
#' Data frame of Denmark's 1385 polling places
#'
#' A dataset containing the longitude and latitude of all of Denmark's 1385 polling places
#'
#' @format A data frame with 155072 rows and 9 variables:
#' \describe{
#'   \item{long}{Longitude of polling place}
#'   \item{lat}{Latitude of polling place}
#'   \item{group}{Groups, generated by \code{ggmap:fortify}}
#'   \item{AfstemKod}{Longitude of polling place}
#'   \item{id}{Id of polling place}
#'   \item{KommuneNum}{Polling place municipality number}
#'   \item{KommuneNav}{Polling place municipality name}
#'   \item{OpstilNum}{...}
#'   \item{OpstilNav}{...}
#' }
#' @source \url{Statistics Denmark}
"polling"
#' Data frame of Danish 2011 election results
#'
#' A dataset containing election results from the Danish general election in 2011 at the polling place level for all major parties
#'
#' @format A data frame with 13774 rows and 3 variables:
#' \describe{
#'   \item{id}{Id of polling place}
#'   \item{navn}{Party name}
#'   \item{stemmer}{Votes (in pct)}
#' }
#' @source \url{http://www.dst.dk/valg/Valg1204271/xml/fintal.xml}
"votes"
#' Data frame of municipal burglaries
#'
#' A dataset containing number of burglaries at the municipality level from 2009
#'
#' @format A data frame with 98 rows and 5 variables:
#' \describe{
#'   \item{indbrud}{Number of burglaries}
#'   \item{year}{Year}
#'   \item{kommune}{Municipality name}
#'   \item{KOMNAVN}{Municipality name}
#'   \item{name}{Name}
#' }
#' @source Statistics Denmark
"crime"
#' Data frame of longitude-latitude points of public benches in Copenhagen
#'
#' A dataset containing longitude-latitude points of public benches in Copenhagen
#'
#' @format A data frame with 2124 rows and 2 variables:
#' \describe{
#'   \item{lon}{Longitude}
#'   \item{lat}{Latitude}
#' }
#' @source Open Data Copenhagen
"benches"
